common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* Type Definitions */
+type StaticType = { a: string; b: string };
+type IndexSignatureType = { [key: string]: string };
+type NullableIndexSignatureType = { [key: string]: string | null };
+type UndefinableIndexSignatureType = { [key: string]: string | undefined };
+
+interface NonNullableProps {
+	definitelyString: string;
+	maybeString?: string;
+}
+
+interface NestedIndexProps {
+	nested: {
+		[key: string]: string;
+	};
+}
+
+/* Constants and Variables */
+const a = "a";
+const b = "b";
+const indexedRecordOfStrings: IndexSignatureType = { a: "hello", b: "goodbye" };
+const nullableIndexedRecord: NullableIndexSignatureType = { a: "hello", b: null };
+const undefinableIndexedRecord: UndefinableIndexSignatureType = { a: "hello", b: undefined };
+
+/*
+ * Accessing Properties
+ */
+
+/* Nullable index signature */
+nullableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+// @ts-expect-error - This should be handled by tsc, not by this custom lint rule
+nullableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined or null
+nullableIndexedRecord.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+
+/* Undefinable index signature */
+undefinableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+// @ts-expect-error - This should be handled by tsc, not by this custom lint rule
+undefinableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined
+undefinableIndexedRecord.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
```

Comment:
Testing all (or at least most) of the cases applied to `indexedRecordOfStrings` also apply to these types. The expectations are a little different for each; so, they should be tested.
I would be okay with a follow-up item to get the complete coverage if some cases are more problematic compared to others. (The tests cases here are not very useful; the interesting ones (I think) are inferred type assignment and explicit assignment. Recommend adding them all and if you run into trouble with some you can comment out the lines with a TODO follow up comment above.)

Reply:
(I would not be surprised if you feel annoyed by having to shift line numbers around in the test file. That can be mitigated by not having such a huge example file with so many cases. Of course, you'll pay the tax if you break it up now. Your call.)

Reply:
(Alternative to such manual line number maintenance:
1. Update all of the defect comments to say // defect: "...exact expected text of defect..." other comments
2. Change test file to read in the example file and find all matches for `// defect: "([^"]+"` including their line number.
3. Loop through hits and assert those exact defects.

)